src/Marker.js

```diff
@@ -88,6 +99,10 @@ export default class Marker extends Component {
     }
   }
 
+  shouldComponentUpdate() {
```

Comment:
At first this was my approach to try fix performance, that is why I implemented it. However, later I didn't remove it because it because I thought that the marker doesn't need to be updated, unless it is completely replaced.

However for the sake of performance, this `shouldComponentUpdate` is not needed. 

@tomchentw what do you think? I can push a change for it.